docs/proposals/PROPOSAL_E.md

```diff
@@ -59,6 +59,7 @@ Create a new artifact media type to support future use cases where a separate co
 {
   "schemaVersion": 2,
```

Comment:
https://github.com/opencontainers/image-spec/blob/main/schema/image-manifest-schema.json#L8
^ isn't this supposed to indicate schema version? which means we are allowed to make schema changes and move forward? Is this being considered at all? Hope I am making sense here.

Reply:
In theory I think you are correct, but I imagine a lot of implementations are keying off of mediatype vs. this field in determining the schema to use... probably worth a discussion

Reply:
I guess the choices are: 

(1) do we announce and ask implementations to fix said bug (ignoring `schemaVersion` field when there is explicitly one sounds like a unmarshalling-101 bug to me) 

OR

(2) do we guarantee that our changes must forever be forward-compatible with older code/tools, which means `schemaVersion` can be deprecated?

PS: Just stating choices, not making one.

Reply:
We've been keeping it for historical reasons. https://github.com/opencontainers/image-spec/blob/main/manifest.md#image-manifest-property-descriptions

I suppose it could be removed, or made optional, since no older runtimes should be parsing a newer artifact-spec manifest (the manifest isn't for packaging images that would be used by those runtimes).

Reply:
For moving forward, I believe @jdolitsky is right, we would be creating a new media type (there's a version in that field). However, those fields are parsed by registries, so a new media type, or new version number in an otherwise known media type, will be rejected by almost every existing registry out there (they parse the manifest to know the associated blobs, perform validation, and often handle garbage collection) until those registries are upgraded. There's no automatic forward compatibility.

Reply:
This is probably a separate conversation from this PR, but I'm curious about it as well. We probably can't remove this field for the existing manifest types (index & image), but why do we need to keep including it for new manifest types? I think this is an example artifact manifest, so shouldn't we just leave off the `schemaVersion` field altogether? There's no comparable "docker" version of the artifact manifest so it shouldn't cause any compatibility issues by leaving it out.

Reply:
If its really just for compatibility, then I think its safe to remove it for the new type.. right?

Reply:
> "For **this version** of the specification, **this MUST be 2** to ensure backward compatibility with older versions of Docker."

The above statement doesn't hold then for **another version** (> v2s2) of the specification? If so, this field does offer an out for a schema selector instead of deprecation, either to support or not support schema versions, with minimal code changes. Thoughts? Furthermore, in the long-term, "backward compatibility with Docker" seems an artificial constraint for an OCI spec.

Reply:
`"schemaVersion": 3` for sure and subsequent schema changes per community consensus.

For old tools, 
(in pseudocode) ...

```
# this is the skip-unsupported change to be added!
read schema as JSON key-val pairs
schemaVersion = json.keys["schemaVersion"]
if schemaVersion not in [v1, v2s1, v2s2]:
   fail with unsupported-error
process further because supported
```

PS: Wouldn't this be a more acceptable path in reality for folks who are not interested in supporting schema changes but want to fail gracefully instead.

Reply:
sounds reasonable to me, others may have more background on this field

Reply:
From today's call given that the schemaVersion is for historic reasons and carrying this over the new manifest seems like carrying over this behavior might not be needed right now for this new manifest. 
https://github.com/opencontainers/wg-reference-types/pull/73